rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/530_date_ignore_malformed.yml

```diff
@@ -0,0 +1,115 @@
+        refresh: true
+        index: test
+        id: "1"
+        body: { "date_ignored": "2024-02-03T10:12:43.123Z", "date_not_ignored": "2024-02-03T10:12:43.123Z", date_nanos_ignored: "2024-02-03T10:12:43.123456789Z", "date_nanos_not_ignored": "2024-02-03T10:12:43.123456789Z" }
+
+  - do:
+      catch: bad_request
+      index:
+        refresh: true
+        index: test
+        id: "2"
+        body: { "date_ignored": "2024-02-03T10:12:43.123Z", "date_not_ignored": { "key": "a", "value": 10 }, date_nanos_ignored: "2024-02-03T10:12:43.123456789Z", "date_nanos_not_ignored": "2024-02-03T10:12:43.123456789Z" }
+
+  - match: { error.root_cause.0.type: "document_parsing_exception" }
+
+  - do:
+      catch: bad_request
+      index:
+        refresh: true
+        index: test
+        id: "3"
+        body: { "date_ignored": "2024-02-03T10:12:43.123Z", "date_not_ignored": "2024-02-03T10:12:43.123Z", date_nanos_ignored: "2024-02-03T10:12:43.123456789Z", "date_nanos_not_ignored": { "key": "b", "value": 20 } }
+
+  - match: { error.root_cause.0.type: "document_parsing_exception" }
+
+  - do:
+      index:
+        refresh: true
+        index: test
+        id: "1"
+        body: { "date_ignored": { "key": "a", "value": 10 }, "date_not_ignored": "2024-02-03T10:12:43.123Z", date_nanos_ignored: { "key": "b", "value": 20 }, "date_nanos_not_ignored": "2024-02-03T10:12:43.123456789Z" }
+
+  - do:
+      index:
+        refresh: true
+        index: test
+        id: "2"
+        body: { "date_ignored": [ 1, 2, 3 ], "date_not_ignored": "2024-02-03T10:12:43.123Z", date_nanos_ignored: [ 10, 20, 30 ], "date_nanos_not_ignored": "2024-02-03T10:12:43.123456789Z" }
+
+  - do:
+      index:
+        refresh: true
+        index: test
+        id: "3"
+        body: { "date_ignored": [ { "key": "a", "value": 10 }, { "key": "c", "value": 100 } ], "date_not_ignored": "2024-02-03T10:12:43.123Z", date_nanos_ignored: [ { "key": "b", "value": 20 }, { "key": "d", "value": 40 } ], "date_nanos_not_ignored": "2024-02-03T10:12:43.123456789Z" }
+
+  - do:
+      index:
+        refresh: true
+        index: test
+        id: "4"
+        body: { "date_ignored": [ "foo", "bar", "baz" ], "date_not_ignored": "2024-02-03T10:12:43.123Z", date_nanos_ignored: [ -10, -20, -30 ], "date_nanos_not_ignored": "2024-02-03T10:12:43.123456789Z" }
+
+  - do:
+      search:
+        index: test
+        body:
+          query:
+            match_all: { }
+
+  - match: { hits.total.value: 4 }
+
+  - match: { hits.hits.0._source.date_ignored.key: "a" }
+  - match: { hits.hits.0._source.date_ignored.value: 10 }
+  - match: { hits.hits.0._source.date_not_ignored: "2024-02-03T10:12:43.123Z" }
+  - match: { hits.hits.0._source.date_nanos_ignored.key: "b" }
+  - match: { hits.hits.0._source.date_nanos_ignored.value: 20 }
+  - match: { hits.hits.0._source.date_nanos_not_ignored: "2024-02-03T10:12:43.123456789Z" }
+
+  # NOTE: this is interesting...numeric values translate to millis since epoch
```

Comment:
Default format indeed accepts `epoch_millis` so these are not malformed.

Reply:
I need to test a couple of more cases...arrays with valid values and arrays with mixed valid and invalid values.